packages/permissions/src/hooks/use-is-authorized/use-is-authorized.spec.tsx

```diff
@@ -187,4 +208,271 @@ describe('when only one permission matches', () => {
       expect(queryByText('Is authorized: Yes')).toBeInTheDocument();
     });
   });
+
+  describe('if action rights', () => {
+    describe('if can view products', () => {
+      describe('if can not publish products on products group', () => {
+        it('should pass isAuthorized as "false"', () => {
+          const { queryByText } = testRender({
+            allAppliedPermissions: [
+              { name: 'canManageProjectSettings', value: true },
+              { name: 'canViewProducts', value: true },
+            ],
+            allAppliedActionRights: [
+              {
+                group: 'products',
+                name: 'canEditPrices',
+                value: false,
+              },
+            ],
+            demandedPermissions: ['ViewProducts'],
+            demandedActionRights: [
+              { group: 'products', name: 'PublishProducts' },
+            ],
+            shouldMatchSomePermissions: true,
+          });
+
+          expect(queryByText('Is authorized: No')).toBeInTheDocument();
+        });
+      });
+      describe('if can edit prices on products group', () => {
```

Comment:
This is a good test.
This proves that it is on a consumer side to be careful when demanding permissions.

because one can expect `isAuthorized=true` if `actualPermissions.ViewProducts=true` and `actualActionRights.canEditPrices=true`, then you get `isAuthorized=true` if you demand this.

However if this happens, we may have a flaw in how we save action rights.